[BUG] difffsitter compilation fails on Ubuntu/Debian for grammars with C++ sources

**Describe the bug**

If you try to build diffsitter with `cargo build`, you get the follower linker error:

Note: Seen in Github actions, which uses Ubuntu, and tested on aarch64 Debian (via QEMU on an M1 Mac)

**To Reproduce**

Clone the repo on Debian/Ubuntu, try running `cargo build`.

**Expected behavior**
Everything should build normally without errors.

**Log output/screenshots**

```
debian@debian:~/diffsitter$ cargo b
   Compiling diffsitter v0.7.1 (/home/debian/diffsitter)
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.10hsy3n80ijqvi4m.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.11ewwa96u9y23vv2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.12knftlhv38chton.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.13hzyd8385gtqnto.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.142efmrmtsutlfth.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.142qtigqw5fa34d0.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.15z4t4w4bv7zvkgd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.16bozstbwxfm6u6c.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.17noxhu49c211sfn.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.17w1vlb00xfz6qz7.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.182utarbyloywqxk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1aqr2mxng9o29kr2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1ftcid4af59u2ov.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1ght1xao0esg2g1g.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1h3vtfrdea4zr531.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1iqgi9qdsl8nvvrm.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1itsrd57988t3lra.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1jajnwmz7gl2azql.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1o2s8txxysej4e7g.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1otyo9fbknxb34xb.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1ov3ol0bhayg0t72.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1r7j5gg0qlviez39.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1r9kn9sbyo8igosr.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1rkn7gfgyrqkdzni.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1s3aizt59savhkax.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1t4on7csgoqc1ptd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1twnas07778457fb.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1v7pzwoudpinjylw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1xz8scto2o4ya09z.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1zem86q0ohn1tnnb.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.1zp778viwwjveuai.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.22mniyxpswfmpx8u.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.23q0c7sdp0mwpmkq.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.24x97j6dwrw7wbvg.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2512n92fgzpflj7p.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.267wug5qyjfewybu.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.269mlet1jk5stqw2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.26az24s78p17vj26.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.293vir2x2uti3aoz.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.29nhqf9b4wdzqwq0.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2d2zkkukapl4y6v0.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2dti8f46cejf1grd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2dvr3lsddaed606t.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2f6nnwj3g7zyxhk1.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2fhy119aljuaq1sd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2fqf1n92b84cbj8e.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2h1ev5nol4ssuxj5.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2i941ntk5utf2px0.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2lbe868bsq5wy70d.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2lkx4ja1kxwqs3ik.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2m1b3xjv0ramau7h.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2m7e7tl82h7pvop.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2mf1xsys4fd054g8.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2ph28kj0uztrp2yd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2ppc6j6v2aqm3l4f.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2q4hxtvbd8vb3atl.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2q66n9q1v00dhhqx.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2t9dnq4j7lkbf042.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2tolv99nk833l3t6.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2ummuaqzttnmjy99.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2w8wfom4g6nv8e1d.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2x4czl2kubeqyg2i.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2xsj98nmot1bon4p.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2yp5k92ps4qnvl4d.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.2z41cpwmw929o4nw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.30jdp5szqm4oxk60.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.31vutgyxdjnc3f2g.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.32bm5vw3849yx7ii.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.332zxp4wtmgzttkk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.339iwdgaxy8kagz4.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3824bwh1w74s5hxx.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.399gb7nglc1iyxe4.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.39t4ipte5upbxs1c.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3a5xsdhen31ipnus.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3ayv9uqofgbkc95q.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3ewulzyi87qvvjk4.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3g25rk9cbumhbq9k.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3gqoy1y6beshcux7.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3gr09xleatqirgpc.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3i0r9qegcs30ws74.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3ju0g4ivnw3tw6i5.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3kug8h768jgulj96.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3ltibmj8e7ybebib.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3nw2ffxdqzjc1zpg.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3o968all0ex38ix0.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3paztmlginqjjdok.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3pol6ia6joo1fcuw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3pz9g2com2lue5pw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3q00nl7bbch434g2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3qbdsjf1wfiemxv0.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3r328r597rzwmk41.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3rehlcmepntd05sw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3ssejmse3qeynzz.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3to20g680l8hzgtd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3tz5pegg25on7471.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3uzymu6fte746xxk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3v94078j1yp3tpw1.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3w5bfq7itvcbpll8.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3w65h2r476hb96lb.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.3waaha2pfib0aiiz.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.40ltt2qbc8owojap.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4249tsbm0uszgyv3.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.45bqw7l4qq6mtu7c.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.45h4rmxv09e09g18.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4791ur0co0gpnzvu.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.481b2eijoe4dyqft.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.49s27yzss6gshdxo.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4b6ljrxls9bbjnt3.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4bbyuxyflpv5n08x.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4bh72ukt70a72mhu.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4bovvkrrkv19580z.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4d429nf0cdg0y2c3.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4ej64w9vm6aokdqw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4go1xb47oxs47iky.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4iijz4yurkbn4tk9.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4jiec9o6ccy5w6of.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4kkr53z9zz0bo2h9.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4koqce48ukmlsmio.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4l20avs5k5rqfjm3.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4md9t0xtw9lf2oo5.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4n86i6cy6w2483zc.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4nars5fpms7wr6vw.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4ntfivf2wxxzjz5z.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4o36y78h14q3lvqi.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4ob1e3176zqj3j9g.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4qhjxs77ftej8jgq.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4r0f6dabx3w8q6i4.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4r0urddm7fcg8uw7.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4tfape6d3adi8k5g.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4tv82vaoyt7kg48x.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4u8so2m1xw0gijk5.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4vxg3m7pynavliiy.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4z0bhddxo784mij4.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4z35ky2c6cx89mb2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4zgwz26pqzh812wt.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.4zw9xvqekrzczz0q.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.52sf0k1x1v81lyoj.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.52vl4getkq3jd777.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.545mfo4dix212bzh.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.55mxgg9udeisdlvs.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.562l1hvqjryndb8x.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.56e00fxd8jah85qv.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.593y5d5n0pyjqs5t.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.5bax3zjim8g0bxyj.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.5bzafkdobrsxgbj2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.5cn42w69pi6osqd.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.5doxxzlk2po2ph73.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.5g872fmf12xljm4y.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.7rnv106mjirv10m.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.9dtfugz92lf9d1u.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.agasaq7l64szu31.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.c0d9czrrhpaeqz6.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.dir168q0xmp80gq.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.djxalp4i66haidk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.ea277hy7pl6pmh6.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.fy9410kpecdxbxy.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.gbb3zotjvie3aqk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.h9cvrmgm2td6zw5.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.hfiq8c8k6agtisj.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.hlfg0oz7ik4pp25.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.htadn8mari2d7dj.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.i1usatffmy0k2f2.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.j0g0jows12ibsuk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.jpsyxxgadboa2xk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.k2idl49pqjbjt5o.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.kfwkug0calduo8n.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.mmpu6e6yk7ty7gp.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.nhcli4zgpilwdbt.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.p91fcq16ty8v8ym.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.p9ch4i9dq454zzk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.qxtvs6wk9xlw003.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.r6ij4xx97g773o9.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.s81jp9eka4hjtms.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.sl2kdruhgelj54l.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.sm32um56xbqp322.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.up19tk5ui5g32fk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.vfbsumb1vj08rvh.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.xmok3n04urjtfec.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.y8rw30w8ud3rbnh.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.yfsd6rn4katrxdq.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.z3rv5x6e28gwbdj.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.ziyfba89ih97eod.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.zrffgudpejeehzk.rcgu.o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574.ig8iasl38et3sv9.rcgu.o" "-Wl,--as-needed" "-L" "/home/debian/diffsitter/target/debug/deps" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out" "-L" "/home/debian/diffsitter/target/debug/build/tree-sitter-3089f0ff9cddce13/out" "-L" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-ljson" "-lgo" "-lruby-cpp-compile-diffsiter" "-ljava" "-lbash-cpp-compile-diffsiter" "-lhcl-cpp-compile-diffsiter" "-ltypescript" "-locaml-cpp-compile-diffsiter" "-ltsx" "-lcss" "-lcpp-cpp-compile-diffsiter" "-lhcl" "-lbash" "-lrust" "-lphp-cpp-compile-diffsiter" "-lpython-cpp-compile-diffsiter" "-Wl,-Bdynamic" "-lstdc++" "-Wl,-Bstatic" "-lruby" "-lphp" "-lpython" "-locaml" "-lcpp" "-lc_sharp" "/home/debian/diffsitter/target/debug/deps/libpretty_env_logger-bb585d90c8df2503.rlib" "/home/debian/diffsitter/target/debug/deps/libenv_logger-64c7345006a627d1.rlib" "/home/debian/diffsitter/target/debug/deps/libhumantime-8d4313982f40ad0b.rlib" "/home/debian/diffsitter/target/debug/deps/libquick_error-f5339b5953b81510.rlib" "/home/debian/diffsitter/target/debug/deps/libxdg-8f63ec880e0a89d9.rlib" "/home/debian/diffsitter/target/debug/deps/libdirs-fb604ce62818563d.rlib" "/home/debian/diffsitter/target/debug/deps/libdirs_sys-eeba71531acde346.rlib" "/home/debian/diffsitter/target/debug/deps/libstrum-090cfe08a4d423ca.rlib" "/home/debian/diffsitter/target/debug/deps/libphf-ccd5bfd8615ebd67.rlib" "/home/debian/diffsitter/target/debug/deps/libphf_shared-f765583ed4074dd5.rlib" "/home/debian/diffsitter/target/debug/deps/libsiphasher-8e418103afb19697.rlib" "/home/debian/diffsitter/target/debug/deps/libserde_json-da5b101f2beaacf1.rlib" "/home/debian/diffsitter/target/debug/deps/libryu-c83b6f5ed20250a4.rlib" "/home/debian/diffsitter/target/debug/deps/libitoa-ec32d2d2270d46db.rlib" "/home/debian/diffsitter/target/debug/deps/libunicode_segmentation-5c15225f06f56329.rlib" "/home/debian/diffsitter/target/debug/deps/libtree_sitter-fb38ad3d17ef8301.rlib" "/home/debian/diffsitter/target/debug/deps/libconsole-270338aa8a050e37.rlib" "/home/debian/diffsitter/target/debug/deps/libregex-b2f61d604c4d3ddc.rlib" "/home/debian/diffsitter/target/debug/deps/libaho_corasick-d0394ca5d9b870c6.rlib" "/home/debian/diffsitter/target/debug/deps/libmemchr-523bf9e63938c0eb.rlib" "/home/debian/diffsitter/target/debug/deps/libregex_syntax-ffdcd873bf5a38c9.rlib" "/home/debian/diffsitter/target/debug/deps/libonce_cell-b06163f4266f732a.rlib" "/home/debian/diffsitter/target/debug/deps/liblogging_timer-d2cd6947e5372529.rlib" "/home/debian/diffsitter/target/debug/deps/libthiserror-ea09fbd52dcf71f8.rlib" "/home/debian/diffsitter/target/debug/deps/liblog-ddbb6a1ffa64ee7f.rlib" "/home/debian/diffsitter/target/debug/deps/libcfg_if-23b8bb8e45cee01d.rlib" "/home/debian/diffsitter/target/debug/deps/libjson5-ebbab77d492ce413.rlib" "/home/debian/diffsitter/target/debug/deps/libserde-7e1a997830777951.rlib" "/home/debian/diffsitter/target/debug/deps/libpest-630459567d33af4a.rlib" "/home/debian/diffsitter/target/debug/deps/libucd_trie-6ec538c1d63d2b67.rlib" "/home/debian/diffsitter/target/debug/deps/libanyhow-341f9f18b2ec548d.rlib" "/home/debian/diffsitter/target/debug/deps/libclap_complete-543e84697f166943.rlib" "/home/debian/diffsitter/target/debug/deps/libclap-d89f77afe5b30a3a.rlib" "/home/debian/diffsitter/target/debug/deps/libstrsim-5f25ca42efb952fe.rlib" "/home/debian/diffsitter/target/debug/deps/libatty-5b97ba6f6f03ac30.rlib" "/home/debian/diffsitter/target/debug/deps/libunicase-4c63a709c7197299.rlib" "/home/debian/diffsitter/target/debug/deps/libtermcolor-a4440a3d9481b84f.rlib" "/home/debian/diffsitter/target/debug/deps/libtextwrap-676806958ff71e67.rlib" "/home/debian/diffsitter/target/debug/deps/libterminal_size-5786930dff7c7818.rlib" "/home/debian/diffsitter/target/debug/deps/liblibc-823de496db8d9047.rlib" "/home/debian/diffsitter/target/debug/deps/libunicode_width-73591fae18d63abf.rlib" "/home/debian/diffsitter/target/debug/deps/libindexmap-97f5f80d45917771.rlib" "/home/debian/diffsitter/target/debug/deps/libhashbrown-249fc22e8aefa8bd.rlib" "/home/debian/diffsitter/target/debug/deps/libclap_lex-f73ce118c94f17ae.rlib" "/home/debian/diffsitter/target/debug/deps/libos_str_bytes-c0d84cc80571f695.rlib" "/home/debian/diffsitter/target/debug/deps/libbitflags-c5a20feb4672066a.rlib" "/home/debian/diffsitter/target/debug/deps/liblazy_static-c4365fea64834937.rlib" "-Wl,--start-group" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libstd-7d5a620ab77655ab.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libpanic_unwind-2de0d40477269ad1.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libobject-59ae13aa5b85e878.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libmemchr-25c2a8eab25c758c.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libaddr2line-484db03cffcca5b6.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libgimli-77337fc21bd1cca4.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_demangle-8399fd816afb0bb6.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libstd_detect-dbd9d453a2c2347c.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libhashbrown-f5e9f9d7692ae465.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libminiz_oxide-0ae529ce319bd2e0.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libadler-91ebd391b9f3e29e.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-79cdd974e440af93.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libunwind-fb4f5845224d8c3b.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcfg_if-12a6b9815d90b1e4.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/liblibc-b0a59add1f3108e8.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/liballoc-cc0694ace9be0e66.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/librustc_std_workspace_core-5a84de01a2ad7b8c.rlib" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcore-d1ce211496f525c3.rlib" "-Wl,--end-group" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib/libcompiler_builtins-9453083e1ae0beda.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-znoexecstack" "-L" "/home/debian/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/lib" "-o" "/home/debian/diffsitter/target/debug/deps/diffsitter-9ab6316895539574" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro,-znow" "-nodefaultlibs"
  = note: /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libhcl.a(parser.o):(.data.rel.ro.language.4321+0xb8): undefined reference to `tree_sitter_hcl_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libhcl.a(parser.o):(.data.rel.ro.language.4321+0xc0): undefined reference to `tree_sitter_hcl_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libhcl.a(parser.o):(.data.rel.ro.language.4321+0xc8): undefined reference to `tree_sitter_hcl_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libhcl.a(parser.o):(.data.rel.ro.language.4321+0xd0): undefined reference to `tree_sitter_hcl_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libhcl.a(parser.o):(.data.rel.ro.language.4321+0xd8): undefined reference to `tree_sitter_hcl_external_scanner_deserialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libbash.a(parser.o):(.data.rel.ro.language.4844+0xb8): undefined reference to `tree_sitter_bash_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libbash.a(parser.o):(.data.rel.ro.language.4844+0xc0): undefined reference to `tree_sitter_bash_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libbash.a(parser.o):(.data.rel.ro.language.4844+0xc8): undefined reference to `tree_sitter_bash_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libbash.a(parser.o):(.data.rel.ro.language.4844+0xd0): undefined reference to `tree_sitter_bash_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libbash.a(parser.o):(.data.rel.ro.language.4844+0xd8): undefined reference to `tree_sitter_bash_external_scanner_deserialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libruby.a(parser.o):(.data.rel.ro.language.4987+0xb8): undefined reference to `tree_sitter_ruby_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libruby.a(parser.o):(.data.rel.ro.language.4987+0xc0): undefined reference to `tree_sitter_ruby_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libruby.a(parser.o):(.data.rel.ro.language.4987+0xc8): undefined reference to `tree_sitter_ruby_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libruby.a(parser.o):(.data.rel.ro.language.4987+0xd0): undefined reference to `tree_sitter_ruby_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libruby.a(parser.o):(.data.rel.ro.language.4987+0xd8): undefined reference to `tree_sitter_ruby_external_scanner_deserialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libphp.a(parser.o):(.data.rel.ro.language.5163+0xb8): undefined reference to `tree_sitter_php_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libphp.a(parser.o):(.data.rel.ro.language.5163+0xc0): undefined reference to `tree_sitter_php_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libphp.a(parser.o):(.data.rel.ro.language.5163+0xc8): undefined reference to `tree_sitter_php_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libphp.a(parser.o):(.data.rel.ro.language.5163+0xd0): undefined reference to `tree_sitter_php_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libphp.a(parser.o):(.data.rel.ro.language.5163+0xd8): undefined reference to `tree_sitter_php_external_scanner_deserialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libpython.a(parser.o):(.data.rel.ro.language.4668+0xb8): undefined reference to `tree_sitter_python_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libpython.a(parser.o):(.data.rel.ro.language.4668+0xc0): undefined reference to `tree_sitter_python_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libpython.a(parser.o):(.data.rel.ro.language.4668+0xc8): undefined reference to `tree_sitter_python_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libpython.a(parser.o):(.data.rel.ro.language.4668+0xd0): undefined reference to `tree_sitter_python_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libpython.a(parser.o):(.data.rel.ro.language.4668+0xd8): undefined reference to `tree_sitter_python_external_scanner_deserialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libocaml.a(parser.o):(.data.rel.language.5120+0xb8): undefined reference to `tree_sitter_ocaml_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libocaml.a(parser.o):(.data.rel.language.5120+0xc0): undefined reference to `tree_sitter_ocaml_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libocaml.a(parser.o):(.data.rel.language.5120+0xc8): undefined reference to `tree_sitter_ocaml_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libocaml.a(parser.o):(.data.rel.language.5120+0xd0): undefined reference to `tree_sitter_ocaml_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libocaml.a(parser.o):(.data.rel.language.5120+0xd8): undefined reference to `tree_sitter_ocaml_external_scanner_deserialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libcpp.a(parser.o):(.data.rel.ro.language.5509+0xb8): undefined reference to `tree_sitter_cpp_external_scanner_create'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libcpp.a(parser.o):(.data.rel.ro.language.5509+0xc0): undefined reference to `tree_sitter_cpp_external_scanner_destroy'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libcpp.a(parser.o):(.data.rel.ro.language.5509+0xc8): undefined reference to `tree_sitter_cpp_external_scanner_scan'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libcpp.a(parser.o):(.data.rel.ro.language.5509+0xd0): undefined reference to `tree_sitter_cpp_external_scanner_serialize'
          /usr/bin/ld: /home/debian/diffsitter/target/debug/build/diffsitter-1cd35836c2b26ac7/out/libcpp.a(parser.o):(.data.rel.ro.language.5509+0xd8): undefined reference to `tree_sitter_cpp_external_scanner_deserialize'
          collect2: error: ld returned 1 exit status
          
  = help: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorustc-link-libkindname)

error: could not compile `diffsitter` due to previous error
```

**Platform:**
 OS: Linux, specifically Ubuntu/Debian so far.

**Additional context**

I noticed that this seems to be an issue only with grammars that have C++ source files.

--- a/build.rs
+++ b/build.rs
@@ -96,6 +96,17 @@ fn compile_grammar(
     cpp_sources: &[PathBuf],
     output_name: &str,
 ) -> Result<(), cc::Error> {
+    // NOTE: that we have to compile the C sources first because the scanner depends on the parser.
+    // Right now the only C libraries are parsers, so we build them before the C++ files, which
+    // are only scanners. This resolves a linker error we were seeing on Linux.
+    if !c_sources.is_empty() {
+        cc::Build::new()
+            .include(include)
+            .files(c_sources)
+            .warnings(false)
+            .try_compile(output_name)?;
+    }
+
     if !cpp_sources.is_empty() {
         cc::Build::new()
             .cpp(true)
@@ -106,13 +117,6 @@ fn compile_grammar(
             .try_compile(&format!("{}-cpp-compile-diffsiter", &output_name))?;
     }
 
-    if !c_sources.is_empty() {
-        cc::Build::new()
-            .include(include)
-            .files(c_sources)
-            .warnings(false)
-            .try_compile(output_name)?;
-    }
     Ok(())
 }
 
@@ -158,15 +162,16 @@ fn preprocess_compile_info(grammar: &GrammarCompileInfo) -> CompileParams {
     // The directory to the source files
     let dir = grammar.path.join("src");
 
-    // Prepend {grammar-repo}/src path to each file
-    let c_sources: Vec<_> = grammar
-        .c_sources
-        .iter()
-        .map(|&filename| dir.join(filename))
-        .collect();
     let cpp_sources: Vec<_> = grammar
         .cpp_sources
         .iter()
+        // Prepend {grammar-repo}/src path to each file
+        .map(|&filename| dir.join(filename))
+        .collect();
+    let c_sources: Vec<_> = grammar
+        .c_sources
+        .iter()
+        // Prepend {grammar-repo}/src path to each file
         .map(|&filename| dir.join(filename))
         .collect();
 

